### PR TITLE
HL-175: Delete the deprecated `associateEntity` and `disassociateEntity` actions

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
⚠️ **WARNING:** Please, do **not** merge the PR before the confirmation from @stannnous or @akostevich-atlassian on the completion of the corresponding platform feature rollout on Atlassian's side.

### Context

The PR removes deletes the actions deprecated within https://github.com/atlassian-labs/figma-for-jira/pull/235.

### Changes

- **chore:** Deletes the deprecated `associateEntity` and `disassociateEntity` actions.

### Notes

- The codebase contains several modules with `V2` postfix (e.g., `entitiesRouterV2`), which corresponds to the new implementation. While the PR deletes the old implementation it does not rename modules with the `V2` postfix to avoid extra changes and make code review simpler. If needed, please, consider removing the `V2` postfix in a separate PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,7 +165,7 @@ If needed, you could test these APIs directly by mimicking Jira backend.
             "id": "${FILE_KEY}",
             "ari": "NOT_USED"
         },
-        "associateWith": {
+        "associatedWith": {
```
Comment (akostevich-atlassian) [inline README.md:168]: **note:** Fix a mistake in `README.md`

File: src/domain/entities/atlassian-design.ts [hunk 2]
```diff
@@ -17,17 +17,6 @@ export interface AtlassianProviderUser {
 	readonly id: string;
 }
 
-export class AtlassianAssociation {
```
Comment (akostevich-atlassian) [inline src/domain/entities/atlassian-design.ts:20]: **note:** The whole concept of `AtlassianAssociation` is removed from the domain of this app 🥳 

**CC:** @stannnous

File: src/web/middleware/jira/jira-iframe-symmetric-jwt-authentication-middleware.ts
Comment (akostevich-atlassian) [inline src/web/middleware/jira/jira-iframe-symmetric-jwt-authentication-middleware.ts:1]: **note:** This code is not related to the deprecated actions but it is not used and can be safely deleted.

**Review by Roystbeef — APPROVED** [review]
i love delete

**Review by stannnous — APPROVED** [review]

**Pull request merged**